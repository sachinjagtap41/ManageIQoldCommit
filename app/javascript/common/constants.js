export const MAX_LENGTH_NAMES = 24;
export const MAX_LENGTH_DESCRIPTIONS = 128;

const maxLengthIs = __('Maximum length is %s characters.');
const requiredMaxLengthIs = __('Required. Maximum length is %s characters.');
const youHaveReachedMax = __(
  'You have reached the maximum length of %s characters.'
);

export const validation = {
  name: {
    help: sprintf(maxLengthIs, MAX_LENGTH_NAMES),
    requiredMessage: sprintf(requiredMaxLengthIs, MAX_LENGTH_NAMES),
    maxLength: MAX_LENGTH_NAMES,
    maxLengthWarning: sprintf(youHaveReachedMax, MAX_LENGTH_NAMES)
  },
  description: {
    help: sprintf(maxLengthIs, MAX_LENGTH_DESCRIPTIONS),
    maxLength: MAX_LENGTH_DESCRIPTIONS,
    maxLengthWarning: sprintf(youHaveReachedMax, MAX_LENGTH_DESCRIPTIONS)
  }
};
