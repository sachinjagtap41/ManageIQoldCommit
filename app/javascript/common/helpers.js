// todo: expose this from patternfly-react instead?
export const getDisplayName = Component =>
  Component.displayName || Component.name || 'Component';
