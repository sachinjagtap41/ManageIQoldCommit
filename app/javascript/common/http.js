const { http } = window;

export default {
  get(url, headers = {}, params = {}) {
    return http.get(url, {
      transformResponse: e => ({ data: e }),
      headers,
      params
    });
  },
  post(url, data = {}, headers = {}) {
    return http.post(url, data, {
      headers,
      transformResponse: e => ({ data: e })
    });
  }
};
