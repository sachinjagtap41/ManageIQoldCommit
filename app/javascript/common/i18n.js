import { addLocaleData } from 'react-intl';

// hard wired to english for now
import en from 'react-intl/locale-data/en';
import fr from 'react-intl/locale-data/fr';
import es from 'react-intl/locale-data/es';
import ja from 'react-intl/locale-data/ja';
import zh_CN from 'react-intl/locale-data/zh';

class I18n {
  constructor(locale, timezone) {
    zh_CN[0].locale = 'zh_CN';
    this.fallbackIntl = !global.Intl;

    [this.locale] = locale.split('-');
    this.timezone = this.fallbackIntl ? 'UTC' : timezone;

    // this.loaded = false;
    // this.ready = this.init();

    // hard wired for now
    addLocaleData([...en, ...fr, ...es, ...ja, ...zh_CN]);
    this.loaded = true;
  }

  // todo: gather react-intl locale-data from the server asynchronously
  // async init() {
  //   await this.fetchIntl();
  //   addLocaleData(await import(`react-intl/locale-data/${this.locale}`));
  //   this.loaded = true;
  // }

  // async fetchIntl() {
  //   if (this.fallbackIntl) {
  //     global.Intl = await import('intl');
  //     await import(`intl/locale-data/jsonp/${this.locale}`);
  //   }
  // }
}

const [htmlElemnt] = document.getElementsByTagName('html');
const langAttr = htmlElemnt.getAttribute('lang') || 'en';
const timezoneAttr = htmlElemnt.getAttribute('data-timezone') || 'UTC';

const i18n = new I18n(langAttr, timezoneAttr);

export default i18n;
