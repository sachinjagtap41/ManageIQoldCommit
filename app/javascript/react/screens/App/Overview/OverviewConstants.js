// reminder... we need to go back and prefix these: V2V_
export const SHOW_MAPPING_WIZARD = 'SHOW_MAPPING_WIZARD';
export const HIDE_MAPPING_WIZARD = 'HIDE_MAPPING_WIZARD';
export const MAPPING_WIZARD_EXITED = 'MAPPING_WIZARD_EXITED';
export const SHOW_PLAN_WIZARD = 'SHOW_PLAN_WIZARD';
export const HIDE_PLAN_WIZARD = 'HIDE_PLAN_WIZARD';
export const PLAN_WIZARD_EXITED = 'PLAN_WIZARD_EXITED';
export const V2V_FETCH_CLUSTERS = 'FETCH_V2V_CLUSTERS';
export const CONTINUE_TO_PLAN = 'CONTINUE_TO_PLAN';
export const V2V_SET_MIGRATIONS_FILTER = 'V2V_SET_MIGRATIONS_FILTER';
export const V2V_RETRY_MIGRATION = 'V2V_RETRY_MIGRATION';
export const SHOW_CONFIRM_MODAL = 'SHOW_CONFIRM_MODAL';
export const HIDE_CONFIRM_MODAL = 'HIDE_CONFIRM_MODAL';
export const ARCHIVE_TRANSFORMATION_PLAN = 'ARCHIVE_TRANSFORMATION_PLAN';

export const FETCH_PROVIDERS = 'FETCH_PROVIDERS';
export const FETCH_NETWORKS = 'FETCH_NETWORKS';
export const FETCH_DATASTORES = 'FETCH_DATASTORES';
export const FETCH_PROVIDERS_URL =
  '/api/providers?expand=resources&attributes=type&filter[]=or%20type=ManageIQ::Providers::Redhat::InfraManager&filter[]=type=ManageIQ::Providers::Vmware::InfraManager';

export const FETCH_V2V_TRANSFORMATION_MAPPINGS =
  'FETCH_V2V_TRANSFORMATION_MAPPINGS';

export const FETCH_V2V_TRANSFORMATION_PLANS = 'FETCH_V2V_TRANSFORMATION_PLANS';
export const FETCH_V2V_ARCHIVED_TRANSFORMATION_PLANS =
  'FETCH_V2V_ARCHIVED_TRANSFORMATION_PLANS';

export const CREATE_V2V_TRANSFORMATION_PLAN_REQUEST =
  'CREATE_V2V_TRANSFORMATION_PLAN_REQUEST';

export const FETCH_V2V_ALL_REQUESTS_WITH_TASKS =
  'FETCH_V2V_ALL_REQUESTS_WITH_TASKS';
export const FETCH_V2V_ALL_ARCHIVED_PLAN_REQUESTS_WITH_TASKS =
  'FETCH_V2V_ALL_ARCHIVED_PLAN_REQUESTS_WITH_TASKS';

export const SHOW_DELETE_CONFIRMATION_MODAL = 'SHOW_DELETE_CONFIRMATION_MODAL';
export const HIDE_DELETE_CONFIRMATION_MODAL = 'HIDE_DELETE_CONFIRMATION_MODAL';
export const SET_MAPPING_TO_DELETE = 'SET_MAPPING_TO_DELETE';
export const YES_TO_DELETE_AND_HIDE_DELETE_CONFIRMATION_MODAL =
  'YES_TO_DELETE_AND_HIDE_DELETE_CONFIRMATION_MODAL';
export const DELETE_INFRASTRUCTURE_MAPPING = 'DELETE_INFRASTRUCTURE_MAPPING';

export const MIGRATIONS_FILTERS = {
  notStarted: __('Not Started Plans'),
  inProgress: __('In Progress Plans'),
  completed: __('Completed Plans'),
  archived: __('Archived Plans')
};
