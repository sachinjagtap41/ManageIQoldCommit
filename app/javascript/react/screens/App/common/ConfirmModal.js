import React from 'react';
import PropTypes from 'prop-types';
import { Modal, Button, Grid, noop } from 'patternfly-react';

// Note: This is in the process of being moved to `patternfly-react`.
// TODO: Replace this implementation with an import

const ConfirmModal = props => {
  const {
    icon,
    title,
    body,
    cancelButtonLabel,
    confirmButtonLabel,
    onCancel,
    onConfirm,
    disableCancelButton,
    disableConfirmButton,
    ...otherProps
  } = props;
  return (
    <Modal {...otherProps} onHide={onCancel}>
      <Modal.Header>
        <Modal.CloseButton onClick={onCancel} />
        <Modal.Title>{title}</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        {!icon ? (
          body
        ) : (
          <Grid.Row className="show-grid">
            <Grid.Col xsHidden md={2} className="text-right">
              {icon}
            </Grid.Col>
            <Grid.Col xs={12} md={10}>
              {body}
            </Grid.Col>
          </Grid.Row>
        )}
      </Modal.Body>
      <Modal.Footer>
        <Button
          bsStyle="default"
          className="btn-cancel"
          onClick={onCancel}
          disabled={disableCancelButton}
        >
          {cancelButtonLabel}
        </Button>
        <Button
          bsStyle="primary"
          onClick={onConfirm}
          disabled={disableConfirmButton}
        >
          {confirmButtonLabel}
        </Button>
      </Modal.Footer>
    </Modal>
  );
};

ConfirmModal.propTypes = {
  show: PropTypes.bool,
  title: PropTypes.string,
  icon: PropTypes.node,
  body: PropTypes.node,
  cancelButtonLabel: PropTypes.string,
  confirmButtonLabel: PropTypes.string,
  onCancel: PropTypes.func,
  onConfirm: PropTypes.func,
  disableCancelButton: PropTypes.bool,
  disableConfirmButton: PropTypes.bool
};

ConfirmModal.defaultProps = {
  show: false,
  title: __('Confirm'),
  icon: null,
  body: <p>{__('Are you sure?')}</p>,
  cancelButtonLabel: __('Cancel'),
  confirmButtonLabel: __('Yes'),
  onCancel: noop,
  onConfirm: noop
};

export default ConfirmModal;
