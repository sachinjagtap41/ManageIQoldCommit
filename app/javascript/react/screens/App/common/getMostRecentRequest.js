const getMostRecentRequest = requests =>
  requests.reduce(
    (prev, current) => (prev.created_on > current.created_on ? prev : current)
  );

export default getMostRecentRequest;
