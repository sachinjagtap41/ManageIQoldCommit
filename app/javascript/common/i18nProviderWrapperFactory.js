import React from 'react';
import { IntlProvider } from 'react-intl';
import { getDisplayName } from './helpers';
import i18n from './i18n';

const i18nProviderWrapperFactory = initialNow => WrappedComponent => {
  const wrappedName = getDisplayName(WrappedComponent);

  class I18nProviderWrapper extends React.Component {
    state = { i18nLoaded: true };

    render() {
      if (!this.state.i18nLoaded) {
        return <span />;
      }
      return (
        <IntlProvider locale={i18n.locale} initialNow={initialNow}>
          <WrappedComponent {...this.props} />
        </IntlProvider>
      );
    }
  }
  I18nProviderWrapper.displayName = `I18nProviderWrapper(${wrappedName})`;

  return I18nProviderWrapper;
};

export { i18nProviderWrapperFactory };
