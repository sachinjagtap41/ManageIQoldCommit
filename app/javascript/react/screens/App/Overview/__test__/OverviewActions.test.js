import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { mockRequest, mockReset } from '../../../../../common/mockRequests';

import {
  fetchTransformationPlansAction,
  setMigrationsFilterAction
} from '../OverviewActions';
import { requestTransformationPlansData } from '../overview.transformationPlans.fixtures';
import { V2V_SET_MIGRATIONS_FILTER } from '../OverviewConstants';

const middlewares = [thunk, promiseMiddleware()];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

afterEach(() => {
  store.clearActions();
  mockReset();
});

describe('fetchTransformationPlansAction', () => {
  const {
    fetchTransformationPlansUrl,
    response
  } = requestTransformationPlansData;

  test('dispatches PENDING and FULFILLED actions', () => {
    mockRequest({
      method: 'GET',
      url: fetchTransformationPlansUrl,
      params: null,
      status: 200,
      ...response
    });

    return store
      .dispatch(
        fetchTransformationPlansAction({
          url: fetchTransformationPlansUrl,
          archived: false
        })
      )
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });

  test('dispatches PENDING and REJECTED actions', () => {
    mockRequest({
      method: 'GET',
      url: fetchTransformationPlansUrl,
      params: null,
      status: 404,
      ...response
    });

    return store
      .dispatch(
        fetchTransformationPlansAction({
          url: fetchTransformationPlansUrl,
          archived: false
        })
      )
      .catch(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});

describe('setMigrationsFilterAction', () => {
  test('sets up the migrations filter action object', () => {
    const activeFilter = 'Migrations Plans Not Started';
    const result = setMigrationsFilterAction(activeFilter);

    expect(result).toEqual({
      type: V2V_SET_MIGRATIONS_FILTER,
      payload: activeFilter
    });
  });
});
