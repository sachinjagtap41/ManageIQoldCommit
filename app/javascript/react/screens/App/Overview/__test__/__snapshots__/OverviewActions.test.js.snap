// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchTransformationPlansAction dispatches PENDING and FULFILLED actions 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_TRANSFORMATION_PLANS_PENDING",
  },
  Object {
    "type": "FETCH_V2V_TRANSFORMATION_PLANS_FULFILLED",
  },
]
`;
